From e00236e271d9601bab53d9335d7886fad7385bc6 Mon Sep 17 00:00:00 2001
From: piano1029 <46605242+piano1029@users.noreply.github.com>
Date: Thu, 31 Aug 2023 13:44:41 +0200
Subject: [PATCH 01/20] Add goal to make bootloader entries more readable

---
 src/libostree/ostree-sysroot-deploy.c | 27 ++++++++++++++++-----------
 1 file changed, 16 insertions(+), 11 deletions(-)

diff --git a/src/libostree/ostree-sysroot-deploy.c b/src/libostree/ostree-sysroot-deploy.c
index 0cbc1048..6820850f 100644
--- a/src/libostree/ostree-sysroot-deploy.c
+++ b/src/libostree/ostree-sysroot-deploy.c
@@ -1786,11 +1786,11 @@ parse_os_release (const char *contents, const char *split)
 
       *eq = '\0';
       const char *quotedval = eq + 1;
-      char *val = g_shell_unquote (quotedval, NULL);
-      if (!val)
+      char *os_pretty_name = g_shell_unquote (quotedval, NULL);
+      if (!os_pretty_name)
         continue;
 
-      g_hash_table_insert (ret, g_steal_pointer (&line), val);
+      g_hash_table_insert (ret, g_steal_pointer (&line), os_pretty_name);
     }
 
   return ret;
@@ -2002,10 +2002,15 @@ install_deployment_kernel (OstreeSysroot *sysroot, int new_bootversion,
 
   g_autoptr (GHashTable) osrelease_values = parse_os_release (contents, "\n");
   /* title */
-  const char *val = g_hash_table_lookup (osrelease_values, "PRETTY_NAME");
-  if (val == NULL)
-    val = g_hash_table_lookup (osrelease_values, "ID");
-  if (val == NULL)
+  // GOAL
+  // If staged make it say "Update {OSNAME}"
+  // If not staged but index 0 make it say "Start {OSNAME}"
+  // If not staged and index not 0 make it say "Rollback"
+
+  const char *os_pretty_name = g_hash_table_lookup (osrelease_values, "PRETTY_NAME");
+  if (os_pretty_name == NULL)
+    os_pretty_name = g_hash_table_lookup (osrelease_values, "ID");
+  if (os_pretty_name == NULL)
     return glnx_throw (error, "No PRETTY_NAME or ID in /etc/os-release");
 
   g_autofree char *deployment_version = NULL;
@@ -2030,8 +2035,8 @@ install_deployment_kernel (OstreeSysroot *sysroot, int new_bootversion,
   /* XXX The SYSLINUX bootloader backend actually parses the title string
    *     (specifically, it looks for the substring "(ostree"), so further
    *     changes to the title format may require updating that backend. */
-  g_autoptr (GString) title_key = g_string_new (val);
-  if (deployment_version && *deployment_version && !strstr (val, deployment_version))
+  g_autoptr (GString) title_key = g_string_new (os_pretty_name);
+  if (deployment_version && *deployment_version && !strstr (os_pretty_name, deployment_version))
     {
       g_string_append_c (title_key, ' ');
       g_string_append (title_key, deployment_version);
@@ -2055,8 +2060,8 @@ install_deployment_kernel (OstreeSysroot *sysroot, int new_bootversion,
       = g_strconcat (bootprefix, bootcsumdir, "/", kernel_layout->kernel_namever, NULL);
   ostree_bootconfig_parser_set (bootconfig, "linux", boot_relpath);
 
-  val = ostree_bootconfig_parser_get (bootconfig, "options");
-  g_autoptr (OstreeKernelArgs) kargs = ostree_kernel_args_from_string (val);
+  os_pretty_name = ostree_bootconfig_parser_get (bootconfig, "options");
+  g_autoptr (OstreeKernelArgs) kargs = ostree_kernel_args_from_string (os_pretty_name);
 
   if (kernel_layout->initramfs_namever)
     {
-- 
2.35.1

