From 491e87cdc03532af6250eb7b54c3e93a92047edc Mon Sep 17 00:00:00 2001
From: piano1029 <46605242+piano1029@users.noreply.github.com>
Date: Thu, 31 Aug 2023 20:15:48 +0200
Subject: [PATCH 04/15] Fix my beginner C mistakes

---
 src/libostree/ostree-sysroot-deploy.c | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/src/libostree/ostree-sysroot-deploy.c b/src/libostree/ostree-sysroot-deploy.c
index 966cad05..c9ea1602 100644
--- a/src/libostree/ostree-sysroot-deploy.c
+++ b/src/libostree/ostree-sysroot-deploy.c
@@ -2039,11 +2039,11 @@ install_deployment_kernel (OstreeSysroot *sysroot, int new_bootversion,
    *     ! This is completely ignored in this version */
   g_autoptr (GString) title_key = g_string_new ("");
 
-  if (deployment.staged) {
+  if (deployment->staged) {
     g_string_append (title_key, "Update ");
     g_string_append (title_key, os_pretty_name);
   } else {
-    if (deployment.index == 0) {
+    if (deployment->index == 0) {
       g_string_append (title_key, "Start ");
       g_string_append (title_key, os_pretty_name);
     } else {
-- 
2.35.1

