From 850a9fb821e1bbfef9eba7c6f098f44bee28ef3d Mon Sep 17 00:00:00 2001
From: piano1029 <46605242+piano1029@users.noreply.github.com>
Date: Fri, 1 Sep 2023 09:40:56 +0200
Subject: [PATCH 06/20] Allow patch workflow to run on workflow_dispatch

---
 .github/workflows/patch.yml | 1 +
 1 file changed, 1 insertion(+)

diff --git a/.github/workflows/patch.yml b/.github/workflows/patch.yml
index ef0e141f..024a4f60 100644
--- a/.github/workflows/patch.yml
+++ b/.github/workflows/patch.yml
@@ -3,6 +3,7 @@ name: Patch Workflow
 on:
   schedule:
     - cron: "0 0 * * *"
+  workflow_dispatch:
 
 jobs:
   patch:
-- 
2.35.1

